src/components/cesium_map/api/spatial.js

```diff
@@ -0,0 +1,624 @@
+      console.log(bb);
+      if (this.selectedBox !== null) {
+        this.viewer.entities.remove(this.selectedBox);
+      }
+
+      if (this.selectBoxCallback !== null) {
+        try {
+          this.selectBoxCallback(bb);
+        } catch (e) {
+          console.warn("SelectedBox callback failed.");
+        }
+      }
+    } else {
+      console.log("start tracking");
+      this.tracking_info.tracking = true;
+      const _this = this;
+      this.tracking_info.polyline = this.viewer.entities.add({
+        polyline: {
+          positions: new Cesium.CallbackProperty(function () {
+            return _this.tracking_info.positions;
+          }, false),
+          material: _this.tracking_info.color,
+          width: _this.tracking_info.width,
+          clampToGround: true,
+          debugShowBoundingVolume: false,
+        }
+      });
+      this.tracking_info.positions.push(posn);
+    }
+  }
+
+  _trackMovement(movement) {
+    if (this.tracking_info.tracking) {
+      const posn = this.viewer.scene.pickPosition(movement.endPosition);
+      if (posn !== undefined) {
+        this.tracking_info.positions.push(posn);
+      }
+    }
+    if (this.mouseCoordinateCallback !== null) {
+      const posn = this.viewer.scene.pickPosition(movement.endPosition);
+      if (posn !== undefined) {
+        const xyz = Cesium.Cartographic.fromCartesian(posn);
+        const lat = Cesium.Math.toDegrees(xyz.latitude);
+        const lon = Cesium.Math.toDegrees(xyz.longitude);
+        try {
+          this.mouseCoordinateCallback(lon, lat, xyz.height);
+        } catch (e) {
+          console.warn("mouseCoordinateCallback failed.")
+        }
+      }
+    }
+  }
+
+  stopTracking() {
+    this.tracking_info.tracking = false;
+    let xyz = Cesium.Cartographic.fromCartesian(this.tracking_info.positions[0]);
+    let bb = new Cesium.Rectangle(xyz.longitude, xyz.latitude, xyz.longitude, xyz.latitude);
+    for (const i in this.tracking_info.positions) {
+      xyz = Cesium.Cartographic.fromCartesian(this.tracking_info.positions[i]);
+      bb = Cesium.Rectangle.expand(bb, xyz);
+    }
+    this.tracking_info.positions = [];
+    return asDRectangle(bb);
+  }
+
+  // This is a function to add cesium tileset
+  // The string with curly will throw an error
+  // addPointsBySource(assetId) {
+  //   const tileset = this.viewer.scene.primitives.add(new Cesium.Cesium3DTileset({
+  //     url: Cesium.IonResource.fromAssetId(assetId),
+  //     depthFailMaterial: new Cesium.PolylineOutlineMaterialProperty(
+  //       {
+  //         color: Cesium.Color.RED,
+  //       }
+  //     ),
+  //   }));
+  //   tileset.style = new Cesium.Cesium3DTileStyle({
+  //     color: {
+  //       conditions: [
+  //         ["${Classification} === 0", "color('purple')"],
+  //         ["${Classification} === 1", "color('brown')"],
+  //         ["${Classification} === 2", "color('cyan')"],
+  //         ["${Classification} === 3", "color('orange')"],
```

Comment:
This curly-in-string format would throw a warning in CRA. Keep it for right now if we will use the `cesium tileset`.